### PR TITLE
Add pyton wrapper for row conv operator.

### LINKED ISSUE #7555: Need add python wrapper for row_conv_op

### PR DESCRIPTION
**Pull request by pkuyym**
Resolves #7555

### CODE REVIEW

File: python/paddle/v2/fluid/layers/nn.py [hunk 1]
```diff
@@ -1971,3 +1971,55 @@ def sequence_reshape(input, new_dim):
...
+    * :math:`W_{i-j}`: The (i-j)-th row of parameters with shape [1, D].
+
+    More details about row_conv please refer to the paper \
+    (http://www.cs.cmu.edu/~dyogatam/papers/wang+etal.iclrworkshop2016.pdf) and
+    the design document \
+    (https://github.com/PaddlePaddle/Paddle/issues/2228#issuecomment-303903645).
+
+    Args:
+        input (Variable): Input variable, a 2D LoDTensor with shape [T, D].
+        future_context_size (int): Future context size.
```
Comment (qingqing01) [inline python/paddle/v2/fluid/layers/nn.py]: Better give example, assumed future_context_size = 2, the context size of convolution is 3 ....

File: python/paddle/v2/fluid/layers/nn.py [hunk 2]
```diff
@@ -1971,3 +1971,55 @@ def sequence_reshape(input, new_dim):
...
+    Examples:
+        .. code-block:: python
+
+            x = fluid.layers.data(name='x', shape=[16],
+                            dtype='float32', lod_level=1)
+            out = fluid.layers.row_conv(input=x, future_context_size=2)
+    """
+    helper = LayerHelper('row_conv', **locals())
+    dtype = helper.input_dtype()
+    filter_shape = [future_context_size + 1, input.shape[1]]
```
Comment (qingqing01) [inline python/paddle/v2/fluid/layers/nn.py]: Better to check the input.shape.size.

File: python/paddle/v2/fluid/layers/nn.py (hunk 1 above)
Comment (pkuyym) [reply python/paddle/v2/fluid/layers/nn.py]: Thanks, done.

File: python/paddle/v2/fluid/layers/nn.py (hunk 2 above)
Comment (pkuyym) [reply python/paddle/v2/fluid/layers/nn.py]: It seems the exception information is informational enough.

File: python/paddle/v2/fluid/layers/nn.py [hunk 3]
```diff
@@ -1971,3 +1971,56 @@ def sequence_reshape(input, new_dim):
...
+    Parameters with shape [future_context_size + 1, D] will be created. The math
+    equation of row convolution is as following:
+
+    .. math::
+        Out_{i} = \sum_{j = i} ^ {i + \\tau} X_{j} \odot W_{i - j}
+
+    In the above equation:
+
+    * :math:`Out_{i}`: The i-th row of output variable with shape [1, D].
+    * :math:`\\tau`: Future context size. Please note, the shape of
```
Comment (sidgoyal78) [inline python/paddle/v2/fluid/layers/nn.py]: Quick clarification: The value of $\tau$ is future_context_size + 1 right?

File: python/paddle/v2/fluid/layers/nn.py [hunk 4]
```diff
@@ -1971,3 +1971,56 @@ def sequence_reshape(input, new_dim):
         outputs={'Out': [out]},
         attrs={'new_dim': new_dim})
     return out
+
+
+def row_conv(input, future_context_size, param_attr=None, act=None):
+    """Row Conv Operator. This layer will apply lookahead convolution to
+    **input**. The input variable should be a 2D LoDTensor with shape [T, D].
+    Parameters with shape [future_context_size + 1, D] will be created. The math
+    equation of row convolution is as following:
```
Comment (sidgoyal78) [inline python/paddle/v2/fluid/layers/nn.py]: Minor typo: following -> follows

File: python/paddle/v2/fluid/layers/nn.py (hunk 3 above)
Comment (pkuyym) [reply python/paddle/v2/fluid/layers/nn.py]: 91345395-5355-49ee-aa49-ea3773afb587

File: python/paddle/v2/fluid/layers/nn.py (hunk 4 above)
Comment (pkuyym) [reply python/paddle/v2/fluid/layers/nn.py]: Done.

File: python/paddle/v2/fluid/layers/nn.py (hunk 3 above)
Comment (sidgoyal78) [reply python/paddle/v2/fluid/layers/nn.py]: Sorry for this minor and maybe frustrating comment, but I think if $\tau = future_context_size$, then the formula in Line: [1983](https://github.com/PaddlePaddle/Paddle/pull/7575/files#diff-40735b79ee0c67af35d4e2611d3f76f1R1983) should have a $i + \tau + 1$. 

Maybe I misunderstood something?

Comment (pkuyym) [reply python/paddle/v2/fluid/layers/nn.py]: I think it is correct, we execute `future context size + 1` times of sum operations.

Comment (pkuyym) [reply python/paddle/v2/fluid/layers/nn.py]: Should be $i + \tau - i + 1$ == $\tau + 1$ ?

Comment (sidgoyal78) [reply python/paddle/v2/fluid/layers/nn.py]: Thanks.. I missed that

**Pull request merged**